flow/activities/snapshot_activity.go

```diff
@@ -16,21 +16,33 @@ import (
 	"github.com/PeerDB-io/peer-flow/shared"
 )
 
+type SlotSnapshotState struct {
+	snapshotName string
+	signal       connpostgres.SlotSignal
+	connector    connectors.CDCPullConnector
+}
+
+type TxnSnapshotState struct {
+	SnapshotName     string
+	SupportsTIDScans bool
+}
+
 type SnapshotActivity struct {
-	SnapshotConnectionsMutex sync.Mutex
-	SnapshotConnections      map[string]SlotSnapshotSignal
-	Alerter                  *alerting.Alerter
+	SnapshotStatesMutex sync.Mutex
+	SlotSnapshotStates  map[string]SlotSnapshotState
+	TxnSnapshotStates   map[string]*TxnSnapshotState
```

Comment:
```suggestion
	TxSnapshotStates   map[string]TxnSnapshotState
```
Might as well be consistent with SlotSnapshotStates map. Also rename to `TxSnap` since `Tx` is used throughout codebase as transaction abbreviation